### PR TITLE
feat (resolver): Add new endpoint for editing secrets

### PR DESCRIPTION
**Pull request by muzzacode**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/ingest/secret/UpdateSecretResolverTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  @BeforeMethod
+  public void before() throws RemoteInvocationException, URISyntaxException {
+    mockClient = Mockito.mock(EntityClient.class);
+    mockSecretService = Mockito.mock(SecretService.class);
+
+    resolver = new UpdateSecretResolver(mockClient, mockSecretService);
+  }
+
+  private DataHubSecretValue createSecretAspectWithTypeUrn(Urn testUrn) {
```
Comment (pedro93) [inline datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/ingest/secret/UpdateSecretResolverTest.java:52]: Suggestion: Rename helper method to match what it does: `createSecretAspectWithUrn`

File: datahub-graphql-core/src/main/resources/ingestion.graphql [hunk 2]
```diff
@@ -36,6 +36,11 @@ extend type Mutation {
   """
   createSecret(input: CreateSecretInput!): String
 
+  """
+  Update a Secret
+  """
+  updateSecret(input: UpdateSecretInput!): String
```
[the remark this answers is not in the document]
Comment (pedro93) [reply datahub-graphql-core/src/main/resources/ingestion.graphql:42]: I would also expect the create & update endpoints return a DataHubSecret graphQL entity (with the secret itself being encrypted)

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/secret/UpdateSecretResolver.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                      secretUrn,
+                      Set.of(SECRET_VALUE_ASPECT_NAME),
+                      context.getAuthentication());
+              if (!entityClient.exists(secretUrn, context.getAuthentication())
+                  || response == null) {
+                throw new IllegalArgumentException(
+                    String.format("Secret for urn %s doesn't exists!", secretUrn));
+              }
+
+              final DataMap dataMap =
```
Comment (pedro93) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/secret/UpdateSecretResolver.java:59]: Please put this logic in a mapper. See `OwnershipTypeMapper` as an example.

File: datahub-graphql-core/src/main/resources/ingestion.graphql (hunk 2 above)
[the remark this answers is not in the document]
Comment (muzzacode) [reply datahub-graphql-core/src/main/resources/ingestion.graphql:42]: @pedro93 @anshbansal Please let me know, I am kind of confused. Should we go with changing the endpoints returning DataHubSecret for create and update both? Plus, urn should be the other parameter like in other endpoints (updatePost) for updateSecret(..).

[the remark this answers is not in the document]
Comment (anshbansal) [reply datahub-graphql-core/src/main/resources/ingestion.graphql:42]: Do not change this in this PR please. I'll do further review for the PR

**Review by anshbansal — APPROVED** [review]

**Pull request merged**